### ISSUE TITLE
Inconsistent speed behaviour and children getting cut off

### ISSUE THREAD

**Issue by igorp1** [justin-chu/react-fast-marquee#18]
Hi there! I've made a 1 minute video to describe this bug, you can find it below:

- https://www.loom.com/embed/fdd293301866472689c59f077a816c91

The link I've used for the video is also below, if you want to check it out yourself:
 
- https://hq-git-dependabot-npmandyarnreact-fast-marquee-121.duffel.sh
 
Thank you!

**Comment by justin-chu** [comment]
@igorp1 Thanks for the video, I'll take a look. Are you using NextJS?

**Comment by igorp1** [comment]
Hey Justin! Yep, we are on Next `v10.2.3`

**Comment by promptdigital** [comment]
I am also having exact same issue as well. I there any solution for this yet? Also using NextJS

**Comment by eddiemora74** [comment]
The issue that I experience is that it is too fast on load. Once I resize the browser screen, the speed goes back to normal.

**Comment by davidgould6** [comment]
Is there any update on this issue ?

**Comment by novelja** [comment]
Same issue here... Any updates on this?

Also, I only experienced this on iOS and Mac. Could not reproduce it on Linux and Android. Please correct me if someone actually saw the issue on those devices as well. Maybe that can help track the bug.

**Comment by shawninder** [comment]
I'm having a similar issue as well (it's too fast and doesn't respect my setting).

**Comment by shawninder** [comment]
Could this be because the measured container and marquee sizes are used before they are set?

See https://github.com/justin-chu/react-fast-marquee/blob/8dd03310d0d1a924645437868542b3c279405338/src/components/Marquee.tsx#L129
If this line expects the values to have been updated (by `setContainerWidth` and `setMarqueeWidth`) already, I don't think this is how React works.

Perhaps this:
```
if (marqueeRef.current && containerRef.current) {
  setContainerWidth(containerRef.current.getBoundingClientRect().width);
  setMarqueeWidth(marqueeRef.current.getBoundingClientRect().width);
}

if (marqueeWidth < containerWidth) {
```
should instead be this:
```
let cWidth = containerWidth;
let mWidth = marqueeWidth;
if (marqueeRef.current && containerRef.current) {
  cWidth = containerRef.current.getBoundingClientRect().width;
  setContainerWidth(cWidth);
  mWidth = marqueeRef.current.getBoundingClientRect().width;
  setMarqueeWidth(mWidth);
}

if (mWidth < cWidth) {
```

Just a thought.

**Comment by kyleatcolenso** [comment]
A client pointed this out to me today too, on Safari iOS. It's such a pain to reproduce though because it seems so random. One idea I had was to set the speed 100ms after component loads, incase it's a race condition.

``` 
import React, { FC, ReactNode, useEffect, useState } from "react";
import { default as MarqueeBase } from "react-fast-marquee";
import "./styles";

interface MarqueeProps {
  children?: ReactNode;
  className?: string;
  speed?: number;
}

const Marquee: FC<MarqueeProps> = ({
  children,
  speed,
  className,
}): JSX.Element => {
  const [fixedSpeed, setFixedSpeed] = useState(75);

  useEffect(() => {
    if (!speed) return;
    const timeout = setTimeout(() => {
      setFixedSpeed(speed);
    }, 1);
    return () => clearTimeout(timeout);
  }, [speed])
  
  return (
    <MarqueeBase className={className} gradient={false} speed={fixedSpeed}>
      {children}
    </MarqueeBase>
  );
};

export default Marquee;
```

**Comment by justin-chu** [comment]
Fixed in 1.4.0

**Comment by koyama-tagbangers** [comment]
This issue should be reopened as it doesn't appear to be resolved yet.
It occurs particularly frequently in Safari.
